File: src/main/Java/boj_step10/boj5073.java
import java.io.*;
import java.util.StringTokenizer;

public class boj5073 {
    public static void main(String[] args) throws IOException {
        BufferedReader reader = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter writer = new BufferedWriter(new OutputStreamWriter(System.out));
        String line;
        while (!(line = reader.readLine()).equals("0 0 0")) {
            StringTokenizer tokenizer = new StringTokenizer(line);
            int a = Integer.parseInt(tokenizer.nextToken());
            int b = Integer.parseInt(tokenizer.nextToken());
            int c = Integer.parseInt(tokenizer.nextToken());
            int max = Math.max(a, Math.max(b, c));
            if (max * 2 - (a + b + c) >= 0) writer.append("Invalid\n");
            else {
                if (a == b && a == c) writer.append("Equilateral\n");
                else if (a == b || a == c || b == c) writer.append("Isosceles\n");
                else writer.append("Scalene\n");
            }
        }
        writer.close();
    }
}

File: src/main/kotlin/boj_step10/boj5073.kt
import java.io.BufferedReader
import java.io.BufferedWriter
import java.io.InputStreamReader
import java.io.OutputStreamWriter

fun main() {
    val r = BufferedReader(InputStreamReader(System.`in`))
    val w = BufferedWriter(OutputStreamWriter(System.out))
    while (true) {
        var triangle = r.readLine().split(" ").map { it.toInt() }
        if(triangle[0] == 0 && triangle[1] == 0 && triangle[2] == 0) break
        triangle = triangle.sorted()
        if (triangle[0] + triangle[1] <= triangle[2]) w.write("Invalid\n")
        else if (triangle[0] == triangle[1] && triangle[0] == triangle[2] && triangle[1] == triangle[2]) w.write("Equilateral\n")
        else if (triangle[0] == triangle[1] || triangle[0] == triangle[2] || triangle[1] == triangle[2]) w.write("Isosceles\n")
        else w.write("Scalene\n")
    }
    w.close()
}